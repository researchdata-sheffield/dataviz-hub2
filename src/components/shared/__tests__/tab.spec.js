/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import Tab from "../tab";
import { render, fireEvent, waitFor, getByText } from "@testing-library/react";

const data = [
  {
    title: "Source",
    content: `The University of Sheffield - <a href="https://www.sheffield.ac.uk/news">Latest News</a>.`
  },
  {
    title: "Detail",
    content: `This is an example of detail section.`
  },
  {
    title: "License",
    content: `This visualisation is covered by a Creative Commons license (<a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>).`
  }
];

describe("tab component", () => {
  let tabComponent;

  beforeEach(() => {
    tabComponent = render(<Tab data={data} />);
  });

  it("should renders correctly", async () => {
    expect(tabComponent).toMatchSnapshot();

    await waitFor(() => {
      expect(document.querySelector("#react-tabs-1")).toHaveClass(
        "react-tabs__tab-panel--selected"
      );
      expect(document.querySelector("#react-tabs-3")).not.toHaveClass(
        "react-tabs__tab-panel--selected"
      );
      expect(document.querySelector("#react-tabs-5")).not.toHaveClass(
        "react-tabs__tab-panel--selected"
      );
    });
  });

  it("should switch to different tab on click", async () => {
    const detailBtn = await getByText(tabComponent.container, "Detail");

    await fireEvent.click(detailBtn);

    const sourceBtn = await getByText(tabComponent.container, "Source");
    const licenseBtn = await getByText(tabComponent.container, "License");

    await waitFor(() => {
      expect(sourceBtn).not.toHaveClass("react-tabs__tab--selected");
      expect(detailBtn).toHaveClass("react-tabs__tab--selected");
      expect(licenseBtn).not.toHaveClass("react-tabs__tab--selected");
    });
  });
});
