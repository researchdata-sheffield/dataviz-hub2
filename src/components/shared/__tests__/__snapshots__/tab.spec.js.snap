// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab component should renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-16 undefined"
        data="[object Object],[object Object],[object Object]"
        data-rttabs="true"
      >
        <ul
          class="react-tabs__tab-list"
          role="tablist"
        >
          <li
            aria-controls="panel:r0:0"
            aria-disabled="false"
            aria-selected="true"
            class="react-tabs__tab react-tabs__tab--selected"
            data-rttab="true"
            id="tab:r0:0"
            role="tab"
            tabindex="0"
          >
            Source
          </li>
          <li
            aria-controls="panel:r0:1"
            aria-disabled="false"
            aria-selected="false"
            class="react-tabs__tab"
            data-rttab="true"
            id="tab:r0:1"
            role="tab"
          >
            Detail
          </li>
          <li
            aria-controls="panel:r0:2"
            aria-disabled="false"
            aria-selected="false"
            class="react-tabs__tab"
            data-rttab="true"
            id="tab:r0:2"
            role="tab"
          >
            License
          </li>
        </ul>
        <div
          aria-labelledby="tab:r0:0"
          class="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="panel:r0:0"
          role="tabpanel"
        >
          The University of Sheffield - 
          <a
            href="https://www.sheffield.ac.uk/news"
          >
            Latest News
          </a>
          .
        </div>
        <div
          aria-labelledby="tab:r0:1"
          class="react-tabs__tab-panel"
          id="panel:r0:1"
          role="tabpanel"
        >
          This is an example of detail section.
        </div>
        <div
          aria-labelledby="tab:r0:2"
          class="react-tabs__tab-panel"
          id="panel:r0:2"
          role="tabpanel"
        >
          This visualisation is covered by a Creative Commons license (
          <a
            href="https://creativecommons.org/licenses/by-sa/4.0/"
          >
            CC BY-SA 4.0
          </a>
          ).
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-16 undefined"
      data="[object Object],[object Object],[object Object]"
      data-rttabs="true"
    >
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel:r0:0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          id="tab:r0:0"
          role="tab"
          tabindex="0"
        >
          Source
        </li>
        <li
          aria-controls="panel:r0:1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab:r0:1"
          role="tab"
        >
          Detail
        </li>
        <li
          aria-controls="panel:r0:2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          id="tab:r0:2"
          role="tab"
        >
          License
        </li>
      </ul>
      <div
        aria-labelledby="tab:r0:0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="panel:r0:0"
        role="tabpanel"
      >
        The University of Sheffield - 
        <a
          href="https://www.sheffield.ac.uk/news"
        >
          Latest News
        </a>
        .
      </div>
      <div
        aria-labelledby="tab:r0:1"
        class="react-tabs__tab-panel"
        id="panel:r0:1"
        role="tabpanel"
      >
        This is an example of detail section.
      </div>
      <div
        aria-labelledby="tab:r0:2"
        class="react-tabs__tab-panel"
        id="panel:r0:2"
        role="tabpanel"
      >
        This visualisation is covered by a Creative Commons license (
        <a
          href="https://creativecommons.org/licenses/by-sa/4.0/"
        >
          CC BY-SA 4.0
        </a>
        ).
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
